### ISSUE TITLE
CA1853 analyzer false positive

### ISSUE THREAD

**Issue by bill-poole** [dotnet/roslyn-analyzers#6518]
### Description

CA1853 (Unnecessary call to `Dictionary.ContainsKey(key)`) is raised if `Dictionary.Remove` is invoked inside an `if` clause, where the condition is `Dictionary.ContainsKey(key)`. This makes sense if the `ContainsKey` and `Remove` calls are against the same dictionary, but is a false positive if not.

You sometimes want to remove an item from one dictionary if not found in another dictionary.

### Reproduction Steps

```csharp
var dictionary1 = new Dictionary<string, string>();
var dictionary2 = new Dictionary<string, string>();

var id = "id";
if (!dictionary1.ContainsKey(id))
{
  dictionary2.Remove(id);
}
```

### Expected behavior

No CA1853 false positive.

### Actual behavior

The analyzer returns a CA1853 false positive.

### Known Workarounds

Suppress the warning.

**Comment by CollinAlpert** [comment]
I'm having trouble reproducing this. Could you post a compiling reproduction and add the SDK version you are using?

**Comment by bill-poole** [comment]
@CollinAlpert, I've updated the original comment to be a compiling reproduction. My SDK version is 7.0.203.

**Comment by CollinAlpert** [comment]
Thanks. It turns out this is a duplicate of #6377 and is already fixed, that is why I couldn't reproduce it.

**Comment by bill-poole** [comment]
Okay sorry about that. I originally posted this in the dotnet runtime repository because I wasn't aware of this repository, which is why I didn't find the originally reported issue.
